api/spec/src/subscription.tsp

```diff
@@ -0,0 +1,390 @@
+
+  /**
+   * Migrates the subscripiton to the procided version of the plan.
+   */
+  @post
+  @operationId("migrateSubscription")
+  @route("/{subscriptionId}/migrate")
+  migrate(
+    @path subscriptionId: ULID,
+    @body body: {
+      @minValue(1)
+      targetVersion: integer;
+    },
+  ): Subscription | OpenMeter.ConflictError | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  /**
+   * Cancels the subscription.
+   */
+  @post
+  @operationId("cancelSubscription")
+  @route("/{subscriptionId}/cancel")
+  cancel(
+    @path subscriptionId: ULID,
+    @body body: {
+      /**
+       * If not provided the subscription is canceled immediately.
+       */
+      effectiveDate?: DateTime;
+    },
+  ): Subscription | OpenMeter.ConflictError | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  /**
+   * Cancels the scheduled cancelation.
+   */
+  @post
+  @operationId("cancelScheduledCancelation")
+  @route("/{subscriptionId}/cancel-scheduled-cancelation")
+  cancelScheduledCancelation(@path subscriptionId: ULID):
+    | Subscription
+    | OpenMeter.ConflictError
+    | OpenMeter.NotFoundError
+    | OpenMeter.CommonErrors;
+
+  @get
+  @operationId("getPhases")
+  @summary("Get the phases of a subscription")
+  @route("/{subscriptionId}/phases")
+  getPhases(@path subscriptionId: ULID): SubscriptionPhase[] | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  // We have separate APIs for manipulating phases (adding and removing) to limit what aspects of RFC6902 we implement.
+  // This decision can later be overturned.
+
+  @delete
+  @operationId("deleteSubscriptionPhase")
+  @route("/{subscriptionId}/phases/{phaseKey}")
+  deletePhase(@path subscriptionId: ULID, @path phaseKey: Key):
+    | {
+        @statusCode _: 204;
+      }
+    | OpenMeter.NotFoundError
+    | OpenMeter.ConflictError
+    | OpenMeter.CommonErrors;
+
+  @post
+  @operationId("addSubscriptionPhase")
+  @route("/{subscriptionId}/phases")
+  addPhase(
+    @path subscriptionId: ULID,
+    @body phase: {
+      key: Key;
+      rateCards: RateCard[];
+      activeFrom: DateTime;
+    },
+  ): SubscriptionPhase | OpenMeter.NotFoundError | OpenMeter.ConflictError | OpenMeter.CommonErrors;
+
+  /**
+   * Extend has to delay both the current and all subsequent phases by the provided duration.
+   */
+  @post
+  @operationId("extendSubscriptionPhase")
+  @route("/{subscriptionId}/phases/{phaseKey}/extend")
+  extendPhase(
+    @path subscriptionId: ULID,
+    @path phaseKey: Key,
+    @body body: {
+      @encode(DurationKnownEncoding.ISO8601)
+      extendBy: duration;
+    },
+  ): SubscriptionPhase[] | OpenMeter.ConflictError | OpenMeter.CommonErrors;
+
+  // Direct management of a SubscriptionPhase's SubscriptionItems would be possible via these APIs.
+  // Its unclear if we'll need them.
+
+  @get
+  @operationId("getSubscriptionPhaseItemsAt")
+  @route("/{subscriptionId}/phases/{phaseKey}/items")
```

Comment:
Do we have use cases where we need to get multiple phases? wondering if list with phase key filter is more flexible

Reply:
tbh I'm not sure if these endpoints are necessary at all, I don't really see why someone wouldn't just use `get` with `expand=true`